monai/handlers/wandb_handlers.py

```diff
@@ -0,0 +1,314 @@
+    def __init__(
+        self,
+        iteration_log: bool = True,
+        epoch_log: bool = True,
+        epoch_event_writer: Callable[[Engine, Any], Any] | None = None,
+        epoch_interval: int = 1,
+        iteration_event_writer: Callable[[Engine, Any], Any] | None = None,
+        iteration_interval: int = 1,
+        output_transform: Callable = lambda x: x[0],
+        global_epoch_transform: Callable = lambda x: x,
+        state_attributes: Sequence[str] | None = None,
+        tag_name: str = DEFAULT_TAG,
+    ):
+        """
+        Args:
+            iteration_log: whether to write data to Weights & Biases when iteration completed,
+                default to `True`.
+            epoch_log: whether to write data to Weights & Biases when epoch completed, default to
+                `True`.
+            epoch_event_writer: customized callable Weights & Biases writer for epoch level. Must
+                accept parameter "engine" and "summary_writer", use default event writer if None.
+            epoch_interval: the epoch interval at which the epoch_event_writer is called.
+                Defaults to 1.
+            iteration_event_writer: customized callable Weights & Biases writer for iteration level.
+                Must accept parameter "engine" and "summary_writer", use default event writer if
+                None.
+            iteration_interval: the iteration interval at which the `iteration_event_writer` is
+                called. Defaults to 1.
+            output_transform: a callable that is used to transform the `ignite.engine.state.output`
+                into a scalar to plot, or a dictionary of {key: scalar}. In the latter case, the
+                output string will be formatted as key: value. By default this value plotting
+                happens when every iteration completed. The default behavior is to print loss from
+                output[0] as output is a decollated list and we replicated loss value for every item
+                of the decollated list. `engine.state` and `output_transform` inherit from the ignite
+                concept: https://pytorch.org/ignite/concepts.html#state, explanation and usage
+                example are in the tutorial:
+                https://github.com/Project-MONAI/tutorials/blob/master/modules/batch_output_transform.ipynb.
+            global_epoch_transform: a callable that is used to customize global epoch number. For
+                example, in evaluation, the evaluator engine might want to use trainer engines epoch
+                number when plotting epoch vs metric curves.
+            state_attributes: expected attributes from `engine.state`, if provided, will extract them
+                when epoch completed.
+            tag_name: when iteration output is a scalar, tag_name is used to plot, defaults to `'Loss'`.
+        """
+        if wandb.run is None:
+            raise wandb.Error("You must call `wandb.init()` before WandbStatsHandler()")
```

Comment:
Hi @binliunls, since there are 2 different handlers (and potentially a third one in the future as per https://github.com/Project-MONAI/MONAI/pull/6305#issuecomment-1499306151), I think its best if the initialization of the wandb run is left independent of the handlers. Would of course love to know your thoughts as well.

cc: @Nic-Ma